### PR TITLE
Add filtering of userlog based on settings

### PR DESCRIPTION
**Pull request by bastianbeier**
## Description

Add filtering of userlog based on settings. Should be merged after https://github.com/owncloud/ocis/pull/10706

## Related Issue

- Closes https://github.com/owncloud/ocis/issues/10769

## How Has This Been Tested?

- test environment: localhost
- test case 1: unshare space with user that has in-app notifications for space unshared disabled

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: services/userlog/pkg/service/filter.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		settingsId = defaults.SettingUUIDProfileEventSpaceDeleted
+	default:
+		// event that cannot be disabled
+		return users
+	}
+
+	for _, u := range users {
+		enabled, err := getSetting(ctx, ulf.valueClient, u, settingsId)
+		if err != nil {
+			ulf.log.Error().Err(err).Msg("cannot get user event setting")
```
Comment (2403905) [inline services/userlog/pkg/service/filter.go:71]: Please add the user ID and setting ID log.Error().Err(err).Str("userid", userid).Msg...

File: services/userlog/pkg/service/filter.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	log         log.Logger
+	valueClient settingssvc.ValueService
+}
+
+func NewUserlogFilter(l log.Logger, vc settingssvc.ValueService) *UserlogFilter {
+	return &UserlogFilter{log: l, valueClient: vc}
+}
+
+// Filter removes users who should not receive an in-app notifications for the event
+func (ulf UserlogFilter) Filter(ctx context.Context, event events.Event, executant *user.UserId, users []string) []string {
```
Comment (kobergj) [inline services/userlog/pkg/service/filter.go:25]: This method would be called as `filter.Filter`. That stutters. Maybe rename it to `Execute`? So it'd be called like `filter.Execute`.

File: services/userlog/pkg/service/filter.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	user "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	"github.com/cs3org/reva/v2/pkg/events"
+	"github.com/owncloud/ocis/v2/ocis-pkg/log"
+	"github.com/owncloud/ocis/v2/ocis-pkg/middleware"
+	settingssvc "github.com/owncloud/ocis/v2/protogen/gen/ocis/services/settings/v0"
+	"github.com/owncloud/ocis/v2/services/settings/pkg/store/defaults"
+	micrometadata "go-micro.dev/v4/metadata"
+)
+
+type UserlogFilter struct {
```
Comment (kobergj) [inline services/userlog/pkg/service/filter.go:15]: exported struct `UserlogFilter` should have a comment or be unexported

File: services/userlog/pkg/service/filter.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	log         log.Logger
+	valueClient settingssvc.ValueService
+}
+
+func NewUserlogFilter(l log.Logger, vc settingssvc.ValueService) *UserlogFilter {
```
Comment (kobergj) [inline services/userlog/pkg/service/filter.go:20]: exported func `NewUserlogFilter` should have a comment or be unexported

**Review by kobergj — CHANGES_REQUESTED** [review]
some nitpicking as always. Rest looks :+1:

**Comment by bastianbeier** [comment]
Thanks for the review!
Findings should be fixed @2403905 @kobergj

File: changelog/unreleased/enhancement-add-events-setting.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+Enhancement: The user settings was extended by the notifications preferences
```
Comment (mmattel) [inline changelog/unreleased/enhancement-add-events-setting.md:1]:
```suggestion
Enhancement: The user settings are extended by notifications preferences
```

File: changelog/unreleased/enhancement-add-events-setting.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+
+The user settings was extended, that a user can store their notifications preferences
```
Comment (mmattel) [inline changelog/unreleased/enhancement-add-events-setting.md:3]:
```suggestion
Users can now store their notifications preferences in the user settings
```

**Review by kobergj — APPROVED** [review]
Nice :heart:

**Pull request merged**